### DISCUSSION TITLE
kept asking for functions from version 2.2.x

### DISCUSSION THREAD

**Discussion by CaligoLab** [pallets/werkzeug#2901] [Q&A]
While tryong to import Flask, requests, ran into this:
[ImportError while running tensorboard command: cannot import name '_plain_int' from 'werkzeug._internal'].
Tried many different variations of versions of flask and werkzeug - the same issues kept coming: some error with _ _ init _ _, then cannot import name 'check_str_tuple' from 'werkzeug._internal'.
Went to github page for the file and started copying the methods that were missing into my file. One resolved - another was missing. While searching for the missing functions, I gradually went back to the 2.2.x version of the file (https://github.com/pallets/werkzeug/blob/2.2.x/src/werkzeug/_internal.py#L62) and as they just kept coming, I just replaced my whole file with it.
Fixed the problem - now flask is running. Current setup: flask 3.0.0, flask-bcrypt 1.0.1,flask-session 0.8.0, werkzeug 3.0.0 (with the werkzeug/_internal.py from the link above)

**Comment by ThiefMaster** [comment]
Is there a question?

- You should not edit installed package files from flask, werkzeug or any other packages
- Use the latest version of everything, and pin everything your (transitive) dependencies to avoid such problems. Tools like `uv pip compile` or `pip-compile` are your friends here.

**Comment by CaligoLab** [comment]
Yes, the question is - why?
I tried the latest versions, tried a bit older ones. Uninstalled and reinstalled both flask and werkzug many times - nothing helped. Believe me, I wouldn't want to change anything from the packages, but it just didn't work. This is the only way it started to launch. Why? I don't know. So I wrote here to get some clarity of why is it happening and how to fix it without shaman dances.
No, installing the latest versions of flask and werkzeug didn't help, it kept asking for those functions.

**Reply by ThiefMaster** [reply]
> No, installing the latest versions of flask and werkzeug didn't help, it kept asking for those functions.

Provide full tracebacks to see WHAT is actually importing those functions.

**Reply by davidism** [reply]
Sounds like a bug to report to tensorflow or whatever is trying to import them.
